iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -31,18 +40,66 @@ private static ValidationException createInvalidValueException(Validable validab
+        LOGGER.error("{}{}", validable.getMessageHeader(), message);
+    }
+
+    public static void throwExceptionOrLogError(Validable validable, String message, boolean throwException) {
+        throwExceptionOrLogError(validable, message, throwException, Reporter.NO_OP);
+    }
+
+    public static void throwExceptionOrLogError(Validable validable, String message, boolean throwException, Reporter reporter) {
+        if (throwException) {
+            throw new ValidationException(validable, message);
+        }
+        logError(validable, message, reporter);
+    }
+
+    private static void throwExceptionOrLogErrorForInvalidValue(Validable validable, double value, String valueName, boolean throwException, Reporter reporter) {
+        throwExceptionOrLogErrorForInvalidValue(validable, value, valueName, null, throwException, reporter);
+    }
+
+    private static void throwExceptionOrLogErrorForInvalidValue(Validable validable, double value, String valueName, String reason, boolean throwException, Reporter reporter) {
+        if (throwException) {
+            throw createInvalidValueException(validable, value, valueName, reason);
+        }
+        logError(validable, createInvalidValueMessage(value, valueName, reason), reporter);
+    }
+
+    public static ValidationLevel checkActivePowerSetpoint(Validable validable, double activePowerSetpoint, boolean throwException) {
+        return checkActivePowerSetpoint(validable, activePowerSetpoint, throwException, Reporter.NO_OP);
+    }
+
+    public static ValidationLevel checkActivePowerSetpoint(Validable validable, double activePowerSetpoint, boolean throwException, Reporter reporter) {
         if (Double.isNaN(activePowerSetpoint)) {
-            throw createInvalidValueException(validable, activePowerSetpoint, "active power setpoint");
+            throwExceptionOrLogErrorForInvalidValue(validable, activePowerSetpoint, ACTIVE_POWER_SETPOINT, throwException, reporter);
+            return ValidationLevel.STATE_ESTIMATION;
         }
+        return ValidationLevel.LOADFLOW;
+    }
+
+    public static ValidationLevel checkHvdcActivePowerSetpoint(Validable validable, double activePowerSetpoint, boolean throwException) {
+        return checkHvdcActivePowerSetpoint(validable, activePowerSetpoint, throwException, Reporter.NO_OP);
     }
 
-    public static void checkHvdcActivePowerSetpoint(Validable validable, double activePowerSetpoint) {
+    public static ValidationLevel checkHvdcActivePowerSetpoint(Validable validable, double activePowerSetpoint, boolean throwException, Reporter reporter) {
         if (Double.isNaN(activePowerSetpoint)) {
-            throw createInvalidValueException(validable, activePowerSetpoint, "active power setpoint");
+            throwExceptionOrLogErrorForInvalidValue(validable, activePowerSetpoint, ACTIVE_POWER_SETPOINT, throwException, reporter);
+            return ValidationLevel.STATE_ESTIMATION;
         } else if (activePowerSetpoint < 0) {
```

Comment:
A negative value is forbidden for both validation levels, not just for loadflow level.